dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/LuceneQueryDateTimeFormatter.java

```diff
@@ -0,0 +1,253 @@
+                                    + regExMatch.getGroups().get(0).getMatch() + " 23:59:59]");
+                        }
+                    }
+
+                    //if structureVarName or field.getVelocityVarName() has numbers, we need to split the clause to format only dates
+                    if (clause.startsWith(
+                            structureVarName + "." + field.getVelocityVarName().toLowerCase()
+                                    + ":")) {
+                        replace = structureVarName + "." + field.getVelocityVarName()
+                                .toLowerCase() + ":" + replaceDateTimeFormatInClause(
+                                replace.substring(replace.indexOf(":") + 1));
+                    } else {
+                        replace = replaceDateTimeFormatInClause(replace);
+                    }
+
+                    query = query.replace(clause, replace);
+
+                    break;
+                }
+            }
+        }
+
+        matches = RegEX.find(query, "\\[([0-9]*)(\\*+) TO ");
+        for (RegExMatch regExMatch : matches) {
+            query = query.replace("[" + regExMatch.getGroups().get(0).getMatch() + regExMatch.getGroups().get(1).getMatch() + " TO ", "["
+                    + regExMatch.getGroups().get(0).getMatch() + " TO ");
+        }
+
+        matches = RegEX.find(query, " TO ([0-9]*)(\\*+)\\]");
+        for (RegExMatch regExMatch : matches) {
+            query = query.replace(" TO " + regExMatch.getGroups().get(0).getMatch() + regExMatch.getGroups().get(1).getMatch() + "]", " TO "
+                    + regExMatch.getGroups().get(0).getMatch() + "]");
+        }
+
+        matches = RegEX.find(query, "\\[([0-9]*) (TO) ([0-9]*)\\]");
+        if(matches.isEmpty()){
+            matches = RegEX.find(query, "\\[([a-z0-9]*) (TO) ([a-z0-9]*)\\]");
+        }
+        for (RegExMatch regExMatch : matches) {
+            query = query.replace("[" + regExMatch.getGroups().get(0).getMatch() + " TO "
+                    + regExMatch.getGroups().get(2).getMatch() + "]", "["
+                    + replaceDateTimeFormatInClause(regExMatch.getGroups().get(0).getMatch())
+                    + " TO " + replaceDateTimeFormatInClause(
+                    regExMatch.getGroups().get(2).getMatch())
+                    + "]");
+        }
+
+        //https://github.com/elasticsearch/elasticsearch/issues/2980
+        if (query.contains( "/" )) {
+            query = query.replaceAll( "/", "\\\\/" );
+        }
+
+        return query;
+    }
+
+
+    /**
+     * Applies supported lucene format to dates in a query
+     * @param unformattedDate - Date or range of dates to be formatted
+     * @return
+     */
+    static String replaceDateTimeFormatInClause(final String unformattedDate) {
+        String result;
+
+        for (Map.Entry<String, String> pattern: LUCENE_DATE_TIME_FORMAT_PATTERN.entrySet()){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'pattern' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4545506)